src/Workspaces/Core/Portable/Shared/Extensions/ICodeDefinitionFactoryExtensions.cs

```diff
@@ -164,12 +169,44 @@ private static bool TryGetValue(IDictionary<string, ISymbol> dictionary, string
             return false;
         }
 
-        public static IEnumerable<SyntaxNode> CreateAssignmentStatements(
+        public static SyntaxNode CreateThrowArgumentNullExpression(
             this SyntaxGenerator factory,
+            Compilation compilation,
+            IParameterSymbol parameter)
+        {
+            return factory.ThrowExpression(
+                factory.ObjectCreationExpression(
+                    compilation.GetTypeByMetadataName("System.ArgumentNullException"),
```

Comment:
This came up for me elsewhere, so a legitimate question: Do we prefer "System.ArgumentNullException" or typeof(System.ArgumentNullException).FullName? Or don't care?